main.js

```diff
@@ -360,12 +362,16 @@ $(function() {
       showMessage('0 units - can not sell.');
       return;
     }
-    inventory[productIndex] -= 1;
-    towns[currentTown].stockpiles[productIndex] += 1;
-    money += price;
+    
+    const count = ($(this).text() ==="all") ? inventory[productIndex] : 1;
```

Comment:
The spacing here is still incorrect, should be `.text() === 'all'` - and please use single quotes for strings. (For consistency with the rest of this file.)